java/code/src/com/suse/manager/maintenance/factory/ScheduleFactory.java

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright (c) 2020 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package com.suse.manager.maintenance.factory;
+
+import static java.util.Collections.emptySet;
+import static java.util.stream.Collectors.toSet;
+
+import com.redhat.rhn.common.hibernate.HibernateFactory;
+import com.redhat.rhn.domain.user.User;
+
+import com.suse.manager.model.maintenance.MaintenanceCalendar;
+import com.suse.manager.model.maintenance.MaintenanceSchedule;
+
+import org.apache.log4j.Logger;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * {@link HibernateFactory} for Maintenance Windows-related objects.
+ */
+public class ScheduleFactory extends HibernateFactory {
+
+    private static Logger log = Logger.getLogger(ScheduleFactory.class);
+
+    /**
+     * Save a MaintenanceSchedule
+     * @param schedule the schedule
+     */
+    public void save(MaintenanceSchedule schedule) {
+        saveObject(schedule);
+    }
+
+    /**
+     * Remove a {@link MaintenanceSchedule}
+     * @param schedule to remove
+     */
+    public void remove(MaintenanceSchedule schedule) {
+        removeObject(schedule);
+    }
+
+    /**
+     * List schedules by User and Calendar
+     * @param user the User
+     * @param calendar the Calendar
+     * @return the list of Schedules
+     */
+    public List<MaintenanceSchedule> listSchedulesByUserAndCalendar(User user, MaintenanceCalendar calendar) {
```

Comment:
Same as in CalendarFactory: as we are in ScheduleFactory, we do not need the word Schedule in the method name.

Reply:
Makes sense. Changing.